File: June/13-06-2024/Padovan Sequence.cs
//{ Driver Code Starts
// Initial Template for C#

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {
                int N = Convert.ToInt32(Console.ReadLine());// Input for size of array
                Solution obj = new Solution();
                int res = obj.padovanSequence(N);
                Console.Write(res);
                Console.Write("\n");
          }

        }
    }
}
// } Driver Code Ends


//User function Template for C#
class Solution
    {
        //Complete this function
        public int padovanSequence(int n)
        {
             //Your Code Start

        if (n == 0 || n == 1 || n == 2)
        {
            return 1;
        }

        const int MOD = 1000000007;
        int p0 = 1, p1 = 1, p2 = 1;
        int p_current = 0;

        for (int i = 3; i <= n; ++i)
        {
            p_current = (p1 + p0) % MOD;
            p0 = p1;
            p1 = p2;
            p2 = p_current;
            
           
        }

        return p_current;
          //Your Code End
        }
        
    }

File: June/13-06-2024/Padovan Sequence.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main(String args[])
        {
            Scanner sc = new Scanner(System.in);
            int t = sc.nextInt();
            while(t-->0)
                {
                    int n = sc.nextInt();
                    Solution ob = new Solution();
                    System.out.println(ob.padovanSequence(n));
                }
        }
}    
// } Driver Code Ends


//User function Template for Java

class Solution
{
  public int padovanSequence(int n)
    {
          //Your Code Start
        if (n == 0 || n == 1 || n == 2) {
            return 1;
        }

        final int MOD = 1000000007;
        int p0 = 1, p1 = 1, p2 = 1;  
        int p_curr = 0;

        for (int i = 3; i <= n; ++i) {
            p_curr= (p1 + p0) % MOD;
            p0 = p1;
            p1 = p2;
            p2 = p_curr;
        }

        return p_curr;
    //Your Code End
    }
    
}
